backend/qemu.pm

```diff
@@ -1175,7 +1190,7 @@ sub close_pipes ($self) {
         $self->{qmpsocket} = undef;
     }
 
-    $self->SUPER::close_pipes() unless exists $self->{stop_only_qemu} && $self->{stop_only_qemu};
+    $self->SUPER::close_pipes() unless !$closeall && ((exists $self->{stop_only_qemu} && $self->{stop_only_qemu}) || $self->{expected_shutdown});
```

Comment:
This would reduce the number of negations by one:
```suggestion
    $self->SUPER::close_pipes() if $closeall || !((exists $self->{stop_only_qemu} && $self->{stop_only_qemu}) || $self->{expected_shutdown});
```
(I hope I got it right.)

---

Maybe it would also make sense to pass `$closeall` so in case the base class implementation would actually do something with the parameter it would actually work:

```suggestion
    $self->SUPER::close_pipes($closeall) if $closeall || !((exists $self->{stop_only_qemu} && $self->{stop_only_qemu}) || $self->{expected_shutdown});
```